File: libs/power-log-parser/src/lib/parsers/battlegrounds-tavern-prizes-parser.ts
import { GameTag } from '@firestone-hs/reference-data';
import { ActionParser } from '../action-parser';
import { GameEventProvider } from '../game-event';
import { GameEntity, Node, NodeType } from '../models';
import { ParserState, StateType } from '../state/parser-state';

export class BattlegroundsTavernPrizesParser implements ActionParser {
	readonly ParserName = 'BattlegroundsTavernPrizesParser';

	constructor(_parserState: ParserState) {}

	AppliesOnNewNode(_node: Node, _stateType: StateType): boolean {
		return false;
	}

	AppliesOnCloseNode(node: Node, stateType: StateType): boolean {
		return stateType === StateType.PowerTaskList && node.Type === NodeType.GameEntity;
	}

	CreateGameEventProviderFromNew(_node: Node): GameEventProvider[] | null {
		return null;
	}

	CreateGameEventProviderFromClose(node: Node): GameEventProvider[] | null {
		const gameEntity = node.Object as GameEntity;
		return [
			GameEventProvider.Create(
				gameEntity.TimeStamp,
				'GAME_SETTINGS',
				() => ({
					Type: 'GAME_SETTINGS',
					Value: {
						BattlegroundsPrizes:
							gameEntity.GetTag(GameTag.DARKMOON_FAIRE_PRIZES_ACTIVE) === 1,
						BattlegroundsSpells: true,
						BattlegroundsQuests:
							gameEntity.GetTag(GameTag.BACON_QUESTS_ACTIVE) === 1,
						BattlegroundsBuddies:
							gameEntity.GetTag(GameTag.BACON_BUDDY_ENABLED) === 1,
						BattlegroundsTrinkets:
							gameEntity.GetTag(GameTag.BACON_TRINKETS_ACTIVE) === 1,
						BattlegroundsAnomalies: [
							gameEntity.GetTag(GameTag.BACON_GLOBAL_ANOMALY_DBID),
						],
						BattlegroundsTimewarped:
							gameEntity.GetTag(GameTag.BACON_ALT_TAVERN_SYSTEM_ACTIVE) === 1,
					},
				}),
				false,
				node,
			),
		];
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Events/Parsers/BattlegroundsTavernPrizesParser.cs
using HearthstoneReplays.Parser;
using HearthstoneReplays.Parser.ReplayData;
using HearthstoneReplays.Parser.ReplayData.GameActions;
using System.Linq;
using HearthstoneReplays.Enums;
using HearthstoneReplays.Parser.ReplayData.Entities;
using System.Collections.Generic;
using static HearthstoneReplays.Events.CardIds;
using System;
using Action = HearthstoneReplays.Parser.ReplayData.GameActions.Action;

namespace HearthstoneReplays.Events.Parsers
{
    public class BattlegroundsTavernPrizesParser : ActionParser
    {
        private GameState GameState { get; set; }
        private ParserState ParserState { get; set; }

        public BattlegroundsTavernPrizesParser(ParserState ParserState)
        {
            this.ParserState = ParserState;
            this.GameState = ParserState.GameState;
        }

        public bool AppliesOnNewNode(Node node, StateType stateType)
        {
            return false;
        }

        public bool AppliesOnCloseNode(Node node, StateType stateType)
        {
            return stateType == StateType.PowerTaskList
                && node.Type == typeof(GameEntity);
        }

        public List<GameEventProvider> CreateGameEventProviderFromNew(Node node)
        {
            return null;
        }

        public List<GameEventProvider> CreateGameEventProviderFromClose(Node node)
        {
            return new List<GameEventProvider> { GameEventProvider.Create(
                    (node.Object as GameEntity).TimeStamp,
                     "GAME_SETTINGS",
                    () => new GameEvent
                    {
                        Type = "GAME_SETTINGS",
                        Value = new {
                             BattlegroundsPrizes = (node.Object as GameEntity).GetTag(GameTag.DARKMOON_FAIRE_PRIZES_ACTIVE) == 1,
                             BattlegroundsSpells = true, // (node.Object as GameEntity).GetTag(GameTag.BACON_SPELLS_ACTIVE) == 1,
                             BattlegroundsQuests = (node.Object as GameEntity).GetTag(GameTag.BACON_QUESTS_ACTIVE) == 1,
                             BattlegroundsBuddies = (node.Object as GameEntity).GetTag(GameTag.BACON_BUDDY_ENABLED) == 1,
                             BattlegroundsTrinkets = (node.Object as GameEntity).GetTag(GameTag.BACON_TRINKETS_ACTIVE) == 1,
                             BattlegroundsAnomalies = new List<int>() { (node.Object as GameEntity).GetTag(GameTag.BACON_GLOBAL_ANOMALY_DBID) },
                             BattlegroundsTimewarped = (node.Object as GameEntity).GetTag(GameTag.BACON_ALT_TAVERN_SYSTEM_ACTIVE) == 1,
                        },
                    },
                    false,
                    node)
                };
        }
    }
}
